scripts/release_dev.js

```diff
@@ -0,0 +1,111 @@
+    console.error(`🟠 You are at "main". Are you sure you wanna release a dev version here?`);
+    process.exit(1);
+  }
+
+  await checkGitStatus();
+}
+
+async function getNewVersion() {
+  const hour = getDateYYYYMMDDHHMMSS(); // Use hour to be an incremental number
+  const currentVersion = packageJson.version;
+
+  if (currentVersion.includes('-dev.')) {
+    console.log('Bumping exisiting dev...');
+    return currentVersion.replace(/-dev\.(.*)$/, `-dev.${hour}`);
+  }
+
+  console.log('Creating a new dev...');
+  const versionType = process.argv.slice(2)[0]; // major | minor | patch
+  if (!versionType) {
+    console.log('🟠 version type is missing. Make sure to run the script from package.json');
+    process.exit(1);
+  }
+  const versionBase = semver.coerce(currentVersion); // 1.0.0-xxxx -> 1.0.0
+  return semver.inc(versionBase, versionType) + `-dev.${hour}`;
+}
+
+async function bumpVersion({ newVersion }) {
+  // The tag will be added on gitCommit()
+  const cmd = `npm version --no-git-tag-version ${newVersion}`;
+  await runExec(cmd);
+}
+
+async function gitCommit({ newVersion }) {
+  console.log('Comitting published version.');
+  const cmd = `git add package.json package-lock.json && git commit -m "Release ${newVersion}" && git tag v${newVersion} && git push && git push origin --tags`;
+  await runExec(cmd);
+}
+
+async function publish({ newVersion, otp }) {
+  console.log('Publishing new version...');
+
+  /*
+    --access=public
+      By default, NPM treats packages with workspace (@remoteoss) as private. This forces it to be public
+    --tag=dev 
+      VERY IMPORTANT: Having a tag tells NPM that this is not a "stable" version,
+      otherwise it will be automatically installed by anyone doing "npm install <package-name>".
+      This forces the devs to be precise in the version with "npm install <package-name>@x.x.x-dev.xxxxx"
+      Know more at: https://stackoverflow.com/a/48038690/4737729
+  */
+  const cmd = `npm publish --access=public --tag=dev --otp=${otp}`;
+  try {
+    await runExec(cmd);
+    console.log(`🎉 Version ${newVersion} published!"`);
+  } catch {
+    console.log('🚨 Publish failed! Perhaps the OTP is wrong.');
+    await revertCommit({ newVersion });
+  }
+}
+
+async function init() {
+  await checkGitBranchAndStatus();
+  const newVersion = await getNewVersion();
+
+  console.log(':: Current version:', packageJson.version);
+  console.log(':::::: New version:', newVersion);
+
+  const answer = await askForConfirmation('Ready to commit and publish it?');
+
+  if (answer === 'no') {
+    process.exit(1);
+  }
+
+  // Check Auth/OTP before the bumpVersion() to reduce the probability
+  // of errors, leading files to be changed/pushed before the actual release.
+  await checkNpmAuth();
+  const otp = await askForText('🔐 What is the NPM Auth OTP? (Check 1PW) ');
+
+  await bumpVersion({ newVersion });
+  await build(); // run after the bumpVersion to have the latest version.
```

Comment:
note: Check my latest commit. https://github.com/remoteoss/json-schema-form/pull/3/commits/28792ddde1d9b12a50c36829f96e9996c941f079 I noticed a "missing feature" where the actual package would be published empty. Why? Because it was missing running the `build` script to generate the `dist/` during the release script.

Reply:
How you can test this (in this repo safely):

- run `npm run release:local` - and unzip the generated tarball. (this is explained in the read the CONTRIBUTING too)

To test the `npm run release:dev:*` or `npm run release:main`

- Comment the line with `await checkGitBranchAndStatus()` to bypass git checks
- Comment the line after the `await build` to ensure it doesn't try to commit or publish.

Reply:
on it now

Reply:
Forgot to say, but I also created the `relese:local` script right before that commit, here: https://github.com/remoteoss/json-schema-form/pull/3/commits/fbadf2ea3039c3fcc0075dad1483a79cd08d360b

Reply:
- `npm run release:local` works lovely and i've tested it in dragon and it shows up all in its glory.
- with the scary code behind comments, the `release:*:*` also are there... 

TBH for me this is good, 🚢 ?

Reply:
Outcome from our Slack huddle:
- The first [dev release](https://www.npmjs.com/package/@remoteoss/json-schema-form) was a success!
- We open an MR in our internal codebase to replace the local imports with the released package.
- We noticed a couple of missing exports : (https://github.com/remoteoss/json-schema-form/pull/3/commits/8116f6fb7947e7c31d770681a5e5c488f8561bd5 and https://github.com/remoteoss/json-schema-form/pull/3/commits/8775cbf454a161de41ebd25b0f237a36e23af76f)

Tomorrow we will:
- Finish replacing the imports
- Wait for a green E2E in our internal codebase
- Merge this branch
- Release beta version in `main` :excited: